data-table/__tests__/table.js

```diff
@@ -4,6 +4,113 @@ beforeEach(() => {
+    beforeEach(() => {
+        table.createDataTable({
+            'wrapperDivId': 'div-id'
+        });
+    });
+
+    test('check the add column button', () => {
+        const config = {
+            'wrapperDivId': 'div-id',
+        };
+        table.createDataTable(config);
+        const contents = document.getElementById(config.wrapperDivId).textContent;
+        expect(contents.substr(0, 12)).toEqual("+ Add column");
+    });
+
+    test('check the default value of rows', () => {
+        const contents = document.getElementsByTagName('table')[0].rows.length;
+        expect(contents).toEqual(4);
+    });
+
+    test('check the default value of columns', () => {
+        const contents = document.getElementsByTagName('table')[0].rows[0].cells.length;
+        expect(contents).toEqual(4);
+    });
+
+    test('properly test the first cell', () => {
+        const contents = document.getElementsByClassName('data-table-cell')[0].textContent;
+        expect(contents).toEqual('Rows');
+    });
+
+    test('properly test the first cell', () => {
+        const contents = document.getElementsByClassName('data-table-cell')[1].textContent;
+        expect(contents).toEqual('Column 1');
+    });
+
+    test('check that subDivs are created', () => {
+        expect(document.getElementsByClassName("SubDiv")).not.toEqual(null);
+    });
+
+    test('check that data-table element is created', () => {
+        expect(document.getElementsByClassName("data-table")).not.toEqual(null);
+    });
+});
+
+describe('basic tests to ensure the buttons can function well', () => {
+    beforeEach(() => {
+        table.createDataTable({
+            'wrapperDivId': 'div-id'
+        });
+    });
+
+    test('check the add column button fuctionality', () => {
+        const addButton = document.getElementsByClassName("add-row-button");
+        let contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(16);
+        addButton[0].click();
+        contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(20);
+    });
+
+    test('check the delete column button fuctionality', () => {
+        const addButton = document.getElementsByClassName("add-row-button");
+        let contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(16);
+        addButton[1].click();
+        contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(12);
+    });
+
+    test('check the add a row button fuctionality', () => {
+        const addButton = document.getElementsByClassName("add-row-button");
+        let contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(16);
+        addButton[2].click();
+        contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(20);
```

Comment:
As above with the magic numbers 16 and 20.

This test seems to be the exact same as `check the add a row button functionality` - would these tests catch if clicking the "add column" button accidentally added a row?

Consider adding a helper function called `getNumRows` and `getNumColumns`, and checking for those - rather than (or in addition to) checking for the number of cells.